vitals/EnumBatteryResolver.java

```diff
@@ -0,0 +1,102 @@
+/**
+ * <copyright>
+ * 
+ * Copyright (c) 2021 ETAS GmbH. All rights reserved.
+ * 
+ * </copyright>
+ */
+package vitals;
+
+import java.util.Collections;
+import java.util.EnumSet;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Set;
+
+/**
+ * 
+ */
+public class EnumBatteryResolver implements BatteryProvider {
+
+	private final Set<? extends Battery> battery;
+	private Map<Float, Battery> temperature;
+	private Map<Float, Battery> stateOfCharge;
+	private final Map<Float, Battery> chargeRate;
+
+	{
+		battery = Collections.unmodifiableSet(EnumSet.allOf(BatteryEnum.class));
+		temperature = new HashMap<>();
+		stateOfCharge = new HashMap<>();
+		chargeRate = new HashMap<>();
+		for (final Battery battery : BatteryEnum.values()) {
+			temperature.put(battery.temperature(), battery);
+			stateOfCharge.put(battery.stateOfCharge(), battery);
+			chargeRate.put(battery.chargeRate(), battery);
+		}
+	}
+
+	@Override
+	public boolean isTemperatureOk(final float temperatureValue) {
+		final Battery batteryTemperature = temperature.get(temperatureValue);
```

Comment:
why don't you parameterize batteryTemperature and limit check functions as an arguments